### PR TITLE
Milestone 1: Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by SaidRasinlic**
### In this milestone I've completed the following things:

- Created Person class with instance variables, constructor, getter & setter, private and public methods.
- Created Student class that extends superclass Person and it's constructor, also added a new method named _play_hooky_.
- Created Teacher class that extends superclass Person and it's constructor, also overrided method named _can_use_services?_.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
```
Comment (TedLivist) [inline person.rb]: - [x] This should be a public method, not private. Kindly remove it from this list of `private` methods.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+    super(name, age, parent_permission: parent_permission)
```
Comment (TedLivist) [inline student.rb]: - [x] Kindly re-organize the parameters here. From the way your Person initialize method is set up, `age` should come before `name`.

  age-stud

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'Unknown', parent_permission: true)
+    super(name, age, parent_permission: parent_permission)
```
Comment (TedLivist) [inline teacher.rb]: - [x] Kindly reorganize your parameters here.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @SaidRasinlic,

Good job so far!
✔️ Good documentation
✔️ Classes are in different files
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (SaidRasinlic) [reply person.rb]: Resolved ✔️

File: student.rb (hunk 2 above)
Comment (SaidRasinlic) [reply student.rb]: Resolved ✔️

File: teacher.rb (hunk 3 above)
Comment (SaidRasinlic) [reply teacher.rb]: Thank you. All required changes are resolved ✔️.

**Review by Lameck1 — APPROVED** [review]
Hi, @SaidRasinlic,

You've done well working on the concerns of the previous review 👍🏼.
The tasks for this milestone are now successfully implemented 💯.

## Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**Please, remember to tag me in your question so I can receive the notification.**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**